src/control/common/proto/ctl/util.go

```diff
@@ -0,0 +1,47 @@
+//
+// (C) Copyright 2020 Intel Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+// The Government's rights to use, modify, reproduce, release, perform, display,
+// or disclose this software are subject to the terms of the Apache License as
+// provided in Contract No. 8F-30005.
+// Any reproduction of computer software, computer software documentation, or
+// portions thereof marked with this legend must also reproduce the markings.
+//
+package ctl
+
+import (
+	"errors"
+
+	"github.com/daos-stack/daos/src/control/common/proto/convert"
+	"github.com/daos-stack/daos/src/control/system"
+)
+
+func (m *SystemQueryReq) SetSystemRanks(sysRanks []system.Rank) error {
```

Comment:
A little counterintuitive to have specific struct methods tossed into a generic util.go. I know there's a similar file for mgmt proto below. But maybe they should be renamed to addons.go or something like that? Methods added onto the autogenerated proto structs.

Reply:
I like that. I didn't put much thought into the filename, clearly. I'll update both of them.